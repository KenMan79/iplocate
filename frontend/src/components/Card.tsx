import styled from "styled-components";

export const Card = styled.div`
  padding: 2rem 1rem;
  margin: 0;
  border-radius: 5px;
  background-color: var(--background-card);

  @media screen and (max-width: 450px) {
    padding: 1rem;
  }
`;

export const CardHeader = styled.h2`
  margin: 0;
  padding: 0;
  color: var(--text-primary);
  font-size: 1.8rem;
`;
