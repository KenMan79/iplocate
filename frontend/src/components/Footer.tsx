import React from "react";
import styled from "styled-components";

export const Footer: React.FC = () => (
  <Wrapper>
    <div>Â© 2020, Jeff Hackshaw</div>
    <MaxMindNotice>
      This product includes GeoLite2 data created by MaxMind, available from{" "}
      <a href="https://www.maxmind.com">https://www.maxmind.com</a>.
    </MaxMindNotice>
  </Wrapper>
);

const Wrapper = styled.footer`
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: var(--background-card);

  @media screen and (max-width: 450px) {
    flex-flow: column nowrap;
    align-items: flex-start;
    padding: 1rem;
  }
`;

const MaxMindNotice = styled.div`
  color: var(--text-light);
  font-weight: 300;
  font-size: 0.7rem;
  text-align: right;
  padding-top: 1rem;

  @media screen and (max-width: 450px) {
    text-align: left;
  }

  a: {
    color: var(--text-secondary);
  }
`;
