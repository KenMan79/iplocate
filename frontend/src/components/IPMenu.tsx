import React from "react";
import { FaEye, FaEyeSlash, FaTrash } from "react-icons/fa";
import styled from "styled-components";
import { IP } from "types";

interface Props {
  ips: IP[];
  onSetCurrentIp(ip: IP): void;
  onRemoveIp(ip: IP): void;
  onToggleIpVisible(ip: IP): void;
}

export const IPMenu: React.FC<Props> = (props) => {
  const { ips, onSetCurrentIp, onToggleIpVisible, onRemoveIp } = props;

  if (ips.length === 0) return null;

  return (
    <Wrapper>
      {ips.map((ip) => (
        <Row key={ip.traits.ipAddress}>
          <RowText onClick={() => onSetCurrentIp(ip)}>
            {ip.traits.ipAddress}
          </RowText>
          <div>
            <RowAction
              onClick={() => onRemoveIp(ip)}
              aria-label="remove ip address"
            >
              <FaTrash />
            </RowAction>
            {ip.hidden ? (
              <RowAction
                onClick={() => onToggleIpVisible(ip)}
                aria-label="toggle ip visibility on"
              >
                <FaEye />
              </RowAction>
            ) : (
              <RowAction
                onClick={() => onToggleIpVisible(ip)}
                aria-label="toggle ip visibility off"
              >
                <FaEyeSlash />
              </RowAction>
            )}
          </div>
        </Row>
      ))}
    </Wrapper>
  );
};

const RowAction = styled.button`
  border-radius: 50%;
  flex: 0;
  width: 45px;
  height: 45px;
  font-size: 1.2rem;
  margin: 0 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;

  :hover {
    background-color: rgba(0, 0, 0, 0.1);
  }
`;

const RowText = styled.button`
  display: block;
  background: none;
  text-align: left;
  font-size: 1.2rem;
  margin: 0;
  font-family: var(--font-family-mono);
  padding: 0.8rem 0;

  :hover {
    text-decoration: underline;
  }
`;

const Row = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
`;

const Wrapper = styled.div`
  padding-top: 1rem;
  color: var(--text-primary);
  button {
    background: none;
    outline: none;
    border: none;
    cursor: pointer;
  }
`;
