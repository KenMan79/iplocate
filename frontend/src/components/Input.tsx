import styled from "styled-components";

export const Input = styled.input`
  background: #fff;
  border: 2px solid var(--text-lightest);
  border-radius: 5px;
  padding: 1rem;
  font-size: 1.2rem;
  width: 100%;
  color: var(--text-primary);
  font-family: var(--font-family-mono);
`;
