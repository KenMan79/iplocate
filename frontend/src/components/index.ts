export * from "./Layout";
export * from "./Input";
export * from "./CollapsableDialog";
export * from "./Card";
export * from "./IPForm";
export * from "./Map";
export * from "./IPMenu";
export * from "./Loading";
