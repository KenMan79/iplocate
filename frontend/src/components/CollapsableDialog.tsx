import React from "react";
import styled from "styled-components";
import { Card } from "./Card";

interface Props {
  collapsed: boolean;
}

export const CollapsableDialog: React.FC<Props> = (props) => {
  const { collapsed, children } = props;

  return (
    <Wrapper collapsed={collapsed}>
      <Card>{children}</Card>
    </Wrapper>
  );
};

const Wrapper = styled.div<Props>`
  position: absolute;
  width: 100%;
  height: 100%;
  background: transparent;

  ${Card} {
    z-index: 1000;
    position: absolute;
    min-width: 400px;
    top: ${(props) => (props.collapsed ? "10px" : "50%")};
    left: ${(props) => (props.collapsed ? "10px" : "50%")};
    transform: ${(props) =>
      props.collapsed ? "" : "translateX(-50%) translateY(-50%)"};
    transition: transform 1s ease, top 1s ease, left 1s ease;
    box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
      0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);
  }
`;
