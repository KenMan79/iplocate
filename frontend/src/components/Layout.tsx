import React from "react";
import styled, { createGlobalStyle } from "styled-components";
import { Footer } from "./Footer";

export const Layout: React.FC = (props) => {
  const { children } = props;

  return (
    <>
      <Global />
      <Wrapper>
        <Content>{children}</Content>
        <Footer />
      </Wrapper>
    </>
  );
};

const Global = createGlobalStyle`
  * {
    box-sizing: border-box;
  }
  html, body {
    padding: 0;
    margin: 0;
  }
`;

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  min-height: 100vh;
  font-family: var(--font-family);
`;

const Content = styled.div`
  flex: 1 1 auto;
  background-color: var(--background-main);
`;
