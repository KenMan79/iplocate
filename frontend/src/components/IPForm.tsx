import React, { useState } from "react";
import styled from "styled-components";
import { Input } from "./Input";
import { FaSearch } from "react-icons/fa";

interface Props {
  onSubmit(value: string): void;
}

// basic validation here, better validation on backend with ajv
// const ipRE = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
const validateIp = (ip: string) => {
  return !!ip;
};

export const IPForm: React.FC<Props> = (props) => {
  const { onSubmit } = props;
  const [value, setValue] = useState("");
  const [error, setError] = useState(false);

  const handleSubmit: React.FormEventHandler = (e) => {
    e.preventDefault();
    if (!validateIp(value)) {
      setError(true);
    } else {
      onSubmit(value);
      setValue("");
    }
  };

  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    setValue(e.currentTarget.value);
    setError(false);
  };

  return (
    <>
      <Form onSubmit={handleSubmit} error={error}>
        <Input
          value={value}
          onChange={handleChange}
          placeholder="IP Address"
          aria-label="IP Address input"
        />
        <button type="submit" aria-label="Search IP Address">
          <FaSearch />
        </button>
      </Form>
      {error && <ErrorMessage>Invalid IP Address.</ErrorMessage>}
    </>
  );
};

const Form = styled.form<{ error: boolean }>`
  margin: 0;
  padding: 0;
  display: flex;
  flex-flow: row nowrap;
  align-items: stretch;
  justify-content: stretch;

  ${Input} {
    flex: 1 1 auto;
    border-right: none;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    border-color: ${(props) =>
      props.error ? "var(--text-error)" : "var(--text-lighter)"};
  }

  ${Input}:focus {
    outline: none;
    border-color: ${(props) =>
      props.error ? "var(--text-error)" : "var(--text-secondary)"};
    border-right: none;
  }

  button {
    flex: 0 0 auto;
    border: none;
    outline: none;
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px;
    background: var(--text-secondary);
    display: flex;
    align-items: center;
    padding: 0 1.2rem;
    font-size: 18px;
    color: #fff;
    cursor: pointer;

    svg {
      display: block;
    }
  }
`;

const ErrorMessage = styled.p`
  color: var(--text-error);
  margin: 0;
  padding: 0.5rem 0 0 0;
`;
