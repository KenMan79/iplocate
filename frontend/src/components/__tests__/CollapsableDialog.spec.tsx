import { CollapsableDialog } from "components/CollapsableDialog";
import React from "react";
import { render } from "test-utils/test-utils";

it("matches snapshot", () => {
  const { asFragment } = render(<CollapsableDialog collapsed />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches collapsed snapshot", () => {
  const { asFragment } = render(<CollapsableDialog collapsed={false} />);
  expect(asFragment()).toMatchSnapshot();
});
