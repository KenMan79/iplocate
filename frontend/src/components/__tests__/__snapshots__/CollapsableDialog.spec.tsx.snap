// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches collapsed snapshot 1`] = `
<DocumentFragment>
  .c2 {
  padding: 2rem 1rem;
  margin: 0;
  border-radius: 5px;
  background-color: var(--background-card);
}

.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  background: transparent;
}

.c0 .c1 {
  z-index: 1000;
  position: absolute;
  min-width: 400px;
  top: 50%;
  left: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
  -webkit-transition: -webkit-transform 1s ease,top 1s ease,left 1s ease;
  -webkit-transition: transform 1s ease,top 1s ease,left 1s ease;
  transition: transform 1s ease,top 1s ease,left 1s ease;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  padding: 2rem 1rem;
  margin: 0;
  border-radius: 5px;
  background-color: var(--background-card);
}

.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
  background: transparent;
}

.c0 .c1 {
  z-index: 1000;
  position: absolute;
  min-width: 400px;
  top: 10px;
  left: 10px;
  -webkit-transition: -webkit-transform 1s ease,top 1s ease,left 1s ease;
  -webkit-transition: transform 1s ease,top 1s ease,left 1s ease;
  transition: transform 1s ease,top 1s ease,left 1s ease;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    />
  </div>
</DocumentFragment>
`;
